README.md

```diff
@@ -42,4 +106,24 @@ make bdd testfile=bdd/spec/001_event_spec.sh
 To run a single test group use the line number corresponding to a `Describe`:
 ```
 make bdd testfile=bdd/spec/001_event_spec.sh:30
-```
\ No newline at end of file
+```
+
+## License
+
+The Alan Programming Language is made up of multiple sub-projects housed within this monorepo. Each subdirectory has its own license file and the project as a whole uses two licenses: The Apache 2.0 license and the Affero GPL 3.0 license, with the breakdown as follows:
+
+* Apache 2.0
+  * bdd
+  * js-runtime
+  * std
+* AGPL 3.0
+  * compiler
+  * runtime
+
+The Apache 2.0 license is freely combinable with the GPL 3 series of licenses as well as with proprietary software, so the standard library and Javascript runtime shim library are freely combinable with your own software projects without any requirement to open source it. It also provides a patent indemnification clause: we grant you a license to any applicable patents while using this software in your own code, so there is no risk of being sued for using it.
+
+The AGPL 3.0 license requires that any changes to the code are published and publicly accessible. This is to make sure that any advancements to the compiler and runtime are available to all. The licensing of these tools does *not* affect the licensing of the code they compile or run, however, otherwise any code compiled by GCC would be GPL 3.0 licensed and any Java code running on the OpenJDK would be GPL 2.0 licensed. (The "linking" exception added to the OpenJDK's licensing terms is not needed for Alan as the entire standard library is Apache 2.0 licensed on purpose.)
+
+If one does need to make changes to the compiler or runtime, obeying the terms is as simple as merely forking this repository and making the changes there. You do not even need to try to upstream those changes. If you use the compiler or runtime without changes, all that is required of you is to provide a link to this repository *if you are asked for it.*
+
+We believe this structure will encourage the greatest growth in the ecosystem. Any kind of software, from fully-freely licensed (Apache, BSD, MIT) to reciprocally-freely licensed ([A]GPL 2/3) to proprietary can be developed in the Alan language, and there are no concerns that Alan Technologies will try to apply patent infringement lawsuits on you, while the compiler and primary runtime are reciprocally-freely licensed to ensure advancements from all parties are available to be used.
```

Comment:
Such a long, unrequested explanation would be off putting to me as a developer, unless it was within a license FAQ. I would also not mention suing or any form legal action, even if it is to guarantee that it won't happen, anywhere throughout. Also off putting if they don't fully understand the licensing world and I believe our target user does not. 

Most people wouldn't make the effort to fully understand it either, but just dismiss it. I would also be more explicit about informing the user what GCC is and why it's licensing is relevant. Our target user will probably not know about GCC's license. I didn't. I was thinking about something more concise such as:

The Apache 2.0 license is freely combinable with the GPL 3 series of licenses as well as with proprietary software, so the standard library and Javascript runtime library are freely combinable with your own software projects without any requirement to open source it.

The AGPL 3.0 license requires that any changes to the code are published and publicly accessible. This is to make sure that any advancements to the compiler and runtime are available to all. The licensing of these tools does *not* affect the licensing of the code they compile or run. Similarly the [GCC compiler collection](https://gcc.gnu.org/) is GPL 3.0 licensed, but code compiled by it is not GPL 3.0 licensed.

Reply:
Sounds good.